File: TubePilotAI.Domain/Entities/PromptTemplate.cs
using TubePilotAI.Domain.Enums;

namespace TubePilotAI.Domain.Entities;

public sealed class PromptTemplate
{
    public Guid Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string Category { get; set; } = string.Empty;

    public string? Description { get; set; }

    public string TemplateText { get; set; } = string.Empty;

    public string? SystemMessage { get; set; }

    public string VariablesJson { get; set; } = "[]";

    public PromptTemplateStatus Status { get; set; } = PromptTemplateStatus.Draft;

    public bool IsDefault { get; set; }

    public string CreatedBy { get; set; } = string.Empty;

    public DateTime CreatedAtUtc { get; set; }

    public DateTime? UpdatedAtUtc { get; set; }
}

File: tubepilotaireact.client/src/types/promptTemplate.ts
export type PromptTemplateStatus = 'Draft' | 'Active' | 'Archived' | 0 | 1 | 2;

export interface PromptTemplate {
  id: string;
  name: string;
  category: string;
  description?: string | null;
  templateText: string;
  systemMessage?: string | null;
  variables: string[];
  status: PromptTemplateStatus;
  isDefault: boolean;
  createdBy: string;
  createdAtUtc: string;
  updatedAtUtc?: string | null;
}

export interface PromptTemplatePayload {
  name: string;
  category: string;
  description?: string | null;
  templateText: string;
  systemMessage?: string | null;
  variables: string[];
  status: PromptTemplateStatus;
  isDefault: boolean;
}

export interface CreatePromptTemplatePayload extends PromptTemplatePayload {
  createdBy: string;
}

export interface PromptTemplateFilters {
  search?: string;
  category?: string;
  status?: string;
}
